File: python/ohce/greeter.py
from datetime import datetime


class SystemClock:
    def current_hour(self):
        now = datetime.now()
        return now.hour


class Greeter:
    def __init__(self):
        self.clock = SystemClock()

    def greet(self):
        current_hour = self.clock.current_hour()
        if 6 <= current_hour < 12:
            return "Good morning"
        if 12 <= current_hour <= 19:
            return "Good afternoon"
        if current_hour >= 20 or current_hour < 6:
            return "Good night"

File: cs/Ohce/Greeter.cs
using System.Diagnostics;
using System.Runtime.CompilerServices;

public class Greeter
{
    private readonly OsClock _clock;
    public Greeter()
    {
        _clock = new OsClock();   
    }

    public string Greet()
    {
        int hour = _clock.GetCurrentHour();
        return hour switch
        {
            (>=6) and (< 12) => "Good morning",
            (>=12) and (<= 19) => "Good afternoon",
            // Fun fact: (>=20) and (<6) does not compile because the compiler
            // thinks this statement is unreachable :P
            _ => "Good night",
        };
    }
}